### PR TITLE
pubsub: dynamically create topics and subscriptions in AWS and GCP benchmarks

### LINKED ISSUE #1258: pubsub: create topics and subscriptions dynamically in benchmarks
The current state is inconsistent with how the conformance tests create these resources dynamically.

### PR DESCRIPTION
**Pull request by ijt**
Fixes #1258.

### CODE REVIEW

File: pubsub/gcppubsub/gcppubsub_test.go [hunk 1]
```diff
@@ -139,21 +127,43 @@ func BenchmarkGcpPubSub(b *testing.B) {
...
-	sub := OpenSubscription(ctx, sc, projectID, benchmarkSubscriptionName, nil)
+	subName := fmt.Sprintf("%s-subscription", b.Name())
+	subPath := fmt.Sprintf("projects/%s/subscriptions/%s", projectID, subName)
+	ds, cleanup2, err := createSubscription(ctx, sc, dt, subName, subPath)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer cleanup2()
+	sub := pubsub.NewSubscription(ds, nil)
+
```
Comment (vangent) [inline pubsub/gcppubsub/gcppubsub_test.go]: `defer sub.Shutdown()` ?

File: pubsub/gcppubsub/gcppubsub_test.go [hunk 2]
```diff
@@ -32,27 +32,7 @@ import (
...
-	//    subscription names.
-	topicName         = "test-topic"
-	subscriptionName0 = "test-subscription-1"
-	subscriptionName1 = "test-subscription-2"
-	projectID         = "go-cloud-test-216917"
-
-	benchmarkTopicName        = "benchmark-topic"
-	benchmarkSubscriptionName = "benchmark-subscription"
-)
+const projectID = "go-cloud-test-216917"
```
Comment (vangent) [inline pubsub/gcppubsub/gcppubsub_test.go]: Can you restore some of the comments for this constant, e.g. "This constant captures the projectID that was used during the last --record." ?

File: pubsub/gcppubsub/gcppubsub_test.go (hunk 1 above)
Comment (ijt) [reply pubsub/gcppubsub/gcppubsub_test.go]: Done.

File: pubsub/gcppubsub/gcppubsub_test.go (hunk 2 above)
Comment (ijt) [reply pubsub/gcppubsub/gcppubsub_test.go]: Done.

**Pull request merged**